### ISSUE TITLE
what is the input of genderNet?  and what is the label for the male and female respectively?

### ISSUE THREAD

**Issue by predawnang** [dimitriStoidis/GenGAN#3]
Dear Author

Thanks to your great work.

I notice the variable name you used in the training code is mels, but there is no fft transformation in your dataset code. I think the input of the genderNet is a wav tensor which has shape of (batch, 1, time), am I right?

After some research of your code and librespeech metadata, i guess the label for male is 0, and the label for female is 1.
and then i feed a wav tensor which has the shape (1, 1, time) to the genderNet, that give me a result ([[9.9956e-01, 2.8115e-04]]) which indicates that this wav belongs to a male. But the wav acctrully belongs to a female. so i got confused, i hope you can give me some help. 

Thanks

**Comment by dimitriStoidis** [comment]
Thank you for your interest in our work.

Regarding your question about the GenderNet model, I would like to clarify that the GenderNet model is a stand-alone binary classifier for gender classification from raw audio signals, and is not part of the GenGAN model. It does not involve any FFT in the training process, and the variable name "mels" may have been misleading. 

The labels you mention are correct. To obtain more accurate predictions on the gender of the speaker, I would recommend using the pre-trained model available in `/models/model.ckpt-90_GenderNet.pt`.

Please let me know if you have any further questions or concerns. Thank you.

**Comment by predawnang** [comment]
Dear author, is this pre-trained GenderNet model generalize well on the unseen dataset, like VCTK dataset?

**Comment by dimitriStoidis** [comment]
The generalisation capabilities of GenderNet have only been evaluated on LibriSpeech dataset. However, it would be  interesting to evaluate its performance on other datasets such as VCTK.

**Comment by carlavintila** [comment]
Dear author, I have noticed that the training pipeline for the GenderNet model is not provided in the repository, only the code for evaluating the already trained model. Would you be able to update the repository to include this, or perhaps to explain here how the model was trained?
